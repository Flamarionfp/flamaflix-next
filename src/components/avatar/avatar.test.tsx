import { render, screen } from "@testing-library/react";
import { Avatar } from "./index";

describe("Avatar Test Case", () => {
  const avatarProps = {
    src: "/path",
    alt: "avatar",
  };

  it("should be full rounded", async () => {
    render(<Avatar {...avatarProps} />);

    const imageElement = await screen.findByAltText(avatarProps.alt);

    expect(imageElement).toHaveClass("rounded-full");
  });

  it("renders avatar unchanged", () => {
    const { container } = render(<Avatar {...avatarProps} />);
    expect(container).toMatchSnapshot();
  });
});
