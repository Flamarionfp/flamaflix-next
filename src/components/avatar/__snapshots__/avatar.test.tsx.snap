// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Test Case renders avatar unchanged 1`] = `
<div>
  <img
    alt="avatar"
    class="rounded-full"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fpath&w=96&q=75"
    srcset="/_next/image?url=%2Fpath&w=48&q=75 1x, /_next/image?url=%2Fpath&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</div>
`;
